mantidproject/mantid
# 3D plots in Workbench via right-click menu

**Description of work.**
This PR makes it so that you can create 3D plots in Workbench without using a script, by right-clicking on a workspace in the workspace widget.

**To test:**
Load a workspace.
Right-click it and go to Plot, there should now be a '3D' sub-menu with Surface, Wireframe, and Contour options. Check that each of these produces the correct plot.
Contour and Wireframe plots should have an additional option in the toolbar to change the colour of the lines.
You should be able to right-click Surface plots to change the colorbar scale.
Basically play around and make sure nothing is broken.
Not all of the toolbar options will work correctly, this is addressed in #28537.

Fixes #27470 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
